### PR TITLE
Bitwise/modulo operators raise error with adjacent +/-/@ characters

### PR DESCRIPTION
**Pull request by robverschoor**
### Description

To avoid sequences like &+ or |- to be interpreted as PG operators in case there is no space in between, a space is inserted in ANTLR behind the bitwise operators AND, OR, XOR, NOT (`&, |, ^, ~`) and the modulo operator (`%`)  if the subsequent expression follows the operator immediately (i.e. without any whitespace) and if the first character of the expressions is '+', '-' or '@'. For the NOT operator (`~`), a space is also inserted before it as a preceding '+' or '-' would also cause an error.

Signed-off-by: Rob Verschoor [rcv@amazon.com](mailto:rcv@amazon.com)

### Issues Resolved

BABEL-1033 Bitwise/modulo T-SQL operators raise error without whitespace and with plus/minus/at sign

### Test Scenarios Covered ###
* **Use case based -** Yes

* **Boundary conditions -** N/A

* **Arbitrary inputs -** N/A

* **Negative test cases -** Yes

* **Minor version upgrade tests -** N/A

* **Major version upgrade tests -** N/A

* **Performance tests -** N/A

* **Tooling impact -** N/A

* **Client tests -** N/A

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Sairakan — APPROVED** [review]

File: test/JDBC/input/operator_binary_whitespace-vu-verify.sql [hunk 1]
```diff
@@ -0,0 +1,139 @@
+-- bit AND/OR/XOR
+select 99&-5
```
Comment (forestkeeper) [inline test/JDBC/input/operator_binary_whitespace-vu-verify.sql:2]: Would you mind add a test case to put multiple operator expressions into single select ? Like : 
```
select 99&-5, 99&-5, 99&-5 , 99&-5, 99&-5, 99&-5, 99&-5 , a, b, c from test 
```

File: test/JDBC/input/operator_binary_whitespace-vu-prepare.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+create view v1_bitop as select 99&+2 as c1, 88|-9 as c2, 77^~5 as c3
+go
+create procedure p1_bitop  as select 99&~5|-4
```
Comment (forestkeeper) [inline test/JDBC/input/operator_binary_whitespace-vu-prepare.sql:3]: Would you add a test case that put multiple expressions into procedure declare like : 
```
select 99&+2 as c1, 88|-9 as c2, 77^~5,a,b,c from test_table
```

File: test/JDBC/input/operator_binary_whitespace-vu-verify.sql [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+go
+exec p1_modulo_op
+go
+declare @v int =10%+3 select @v
+go
+declare @v int =10%+~3 select @v
+go
+declare @v int =10%~+3 select @v
+go
+declare @v int = 10 set @v = 3%@v select @v
```
Comment (forestkeeper) [inline test/JDBC/input/operator_binary_whitespace-vu-verify.sql:107]: After declare @v, would you add select @v so we'll be able to test and verify the output

Comment (robverschoor) [reply test/JDBC/input/operator_binary_whitespace-vu-verify.sql:107]: `select @v` is at the end of each line

File: test/JDBC/input/operator_binary_whitespace-vu-prepare.sql (hunk 2 above)
Comment (robverschoor) [reply test/JDBC/input/operator_binary_whitespace-vu-prepare.sql:3]: testcases added

File: test/JDBC/input/operator_binary_whitespace-vu-verify.sql (hunk 1 above)
Comment (robverschoor) [reply test/JDBC/input/operator_binary_whitespace-vu-verify.sql:2]: testcases added

**Review by forestkeeper — APPROVED** [review]

**Pull request merged**